File: D1/AbstractFactory/Equipment.py
class Clothes:
    def __init__(self):
        self.defense = 0

    def display(self):
        print(f"{self.__class__.__name__} defense={self.defense}")

class Armor(Clothes):
    pass

class Leather(Clothes):
    pass

class Weapon:
    def __init__(self):
        self.attack = 0

    def display(self):
        print(f"{self.__class__.__name__} attack={self.attack}")

class Sword(Weapon):
    pass

class Bow(Weapon):
    pass

File: DESIGN-CSharp/AbstractFactory/Equipment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbstractFactoy
{
    public abstract class Clothes
    {
        protected internal int def { get; set; }

        public void display()
        {
            Console.WriteLine(this.GetType().Name + " defense=" + def);
        }
    }

    public class Armor : Clothes
    {
    }

    public class Leather : Clothes
    {
    }

    public abstract class Weapon
    {
        protected internal int atk;

        public void display()
        {
            Console.WriteLine(this.GetType().Name + " attack=" + atk);
        }
    }

    public class Sword : Weapon
    {
    }

    public class Bow : Weapon
    {
    }

}